src/main/java/com/flightstats/hub/webhook/WebhookError.java

```diff
@@ -37,80 +31,42 @@
         limitChildren(webhook);
     }
 
-    private List<String> limitChildren(String webhook) {
-        String errorRoot = getErrorRoot(webhook);
-        List<String> results = new ArrayList<>();
-        SortedMap<String, Error> errors = new TreeMap<>();
-        try {
-            for (String child : curator.getChildren().forPath(errorRoot)) {
-                Stat stat = new Stat();
-                byte[] bytes = curator.getData().storingStatIn(stat).forPath(getChildPath(errorRoot, child));
-                errors.put(child, new Error(child, new DateTime(stat.getCtime()), new String(bytes)));
-            }
-            while (errors.size() > MAX_SIZE) {
-                String firstKey = errors.firstKey();
-                errors.remove(firstKey);
-                curator.delete().inBackground().forPath(getChildPath(errorRoot, firstKey));
-            }
-            DateTime cutoffTime = TimeUtil.now().minusDays(1);
-            for (Error error : errors.values()) {
-                if (error.getCreationTime().isBefore(cutoffTime)) {
-                    curator.delete().inBackground().forPath(getChildPath(errorRoot, error.getName()));
-                } else {
-                    results.add(error.getData());
-                }
-            }
-        } catch (KeeperException.NoNodeException ignore) {
-            logger.debug(ignore.getMessage());
-        } catch (Exception e) {
-            logger.warn("unable to limit children " + errorRoot, e);
-        }
-        return results;
-    }
-
     public void delete(String webhook) {
-        String errorRoot = getErrorRoot(webhook);
-        logger.info("deleting " + errorRoot);
-        try {
-            curator.delete().deletingChildrenIfNeeded().forPath(errorRoot);
-        } catch (KeeperException.NoNodeException e) {
-            logger.info("unable to delete missing node " + errorRoot);
-        } catch (Exception e) {
-            logger.warn("unable to delete " + errorRoot, e);
-        }
+        logger.info("deleting webhook errors for " + webhook);
+        webhookErrorService.deleteWebhook(webhook);
     }
 
-    private String getErrorRoot(String webhook) {
-        return "/GroupError/" + webhook;
+    public List<String> get(String webhook) {
+        return limitChildren(webhook).stream()
+                .map(com.flightstats.hub.webhook.error.WebhookError::getData)
+                .collect(toList());
     }
 
-    private String getChildPath(String errorRoot, String child) {
-        return errorRoot + "/" + child;
-    }
+    private List<com.flightstats.hub.webhook.error.WebhookError> limitChildren(String webhook) {
+        List<com.flightstats.hub.webhook.error.WebhookError> errors = webhookErrorService.getErrors(webhook);
 
-    public List<String> get(String webhook) {
-        return limitChildren(webhook);
+        List<com.flightstats.hub.webhook.error.WebhookError> prunedErrors = webhookErrorPruner.pruneErrors(webhook, errors);
```

Comment:
I dislike having two `WebhookError` classes (one for data in the `webhook.error` package and then this one). If anyone has name change suggestions, that would be great.

Reply:
OK, I renamed some classes. It's better, but not good. 🤷‍♂️